File: js/6-zigzagconversion.js
var convert = function(s, numRows) {
  if (numRows == 1) return s;
  ques = [];
  for (var k = 0; k < numRows; ++k) ques[k]="";
  for(var i = 0; i < s.length; ++i) {
    var j = i % (2*numRows-2) ;
    var cur = j * (j < numRows) + (2 * numRows - 2 - j) * (j >= numRows) ;
    ques[cur] += s[i];
  }
  return ques.join("");
};

File: python/6-zigzagconversion.py
class Solution:
    def convert(self, s, numRows):
        if numRows == 1 : return s 
        ques = [""] * numRows
        for i,c in enumerate(s):
            j = i % (2*numRows-2) ;
            cur = j * (j < numRows) + (2 * numRows - 2 - j) * (j >= numRows) ;
            ques[cur] += c
        return "".join(ques) if numRows > 1 else s
s = Solution()
print s.convert("PAYPALISHIRING", 3)
print s.convert("PAYPALISHIRING", 1) 


